.github/workflows/e2e_trigger_via_label.yaml

```diff
@@ -18,4 +18,10 @@ jobs:
 
     steps:
     - uses: actions/checkout@v3
-    - run: gh workflow run ${WORKFLOW} --ref ${BRANCH} -f pr-number=${PR_NUMBER}
+    - run: |
+        gh workflow run ${WORKFLOW} --ref ${BRANCH} \
+          -f run-gke=true \
+          -f run-istio=true \
+          -f all-supported-k8s-versions=true \
+          -f pr-number=${PR_NUMBER}
+    - run: gh pr edit ${PR_NUMBER} --remove-label ci/run-e2e
```

Comment:
The reason this was placed in the e2e workflow was to have a consistent timeline in PRs. Specifically: you had "add-label", <wait>, comment, "remove-label" events in the timeline.

With this change you'll have "add-label", comment, <wait>, "remove-label"

From UX standpoint I'd prefer to retain the former. WDYT?

Reply:
My motivation to move it here was to make a single workflow responsible for the label itself. When you look at the `e2e_trigger_via_label.yaml` you can see its responsibility is to: 
1) be triggered on `ci/run-e2e` labeled event 
2) run the targeted e2e workflow
3) remove the label

I thought that putting it into the `e2e_targeted` itself is kinda strange as it doesn't mention the label anywhere else but in this step. 

I agree that from the UX perspective it's better this way, so let's make it so: [7ba9471](https://github.com/Kong/kubernetes-ingress-controller/pull/3724/commits/7ba9471b42cefb0779441e4594f00997e1fe3591).